tensorflow/core/lib/gtl/manual_constructor_test.cc

```diff
@@ -92,9 +92,11 @@ TEST(ManualConstructorTest, Alignment) {
 
   EXPECT_EQ(reinterpret_cast<char*>(test2.b.get()) - &test2.a,
             reinterpret_cast<char*>(&control2.b) - &control2.a);
-#ifdef __x86_64__
+
+# if defined __x86_64__ && !defined(PLATFORM_WINDOWS)
```

Comment:
It should be guaranteed, and will cause issues if it isn't.  The `ManualConstructor` does an explicit allocation via malloc.  That should have an alignment according to:

> The storage space pointed to by the return value is suitably aligned for storage of any type of object that has an alignment requirement less than or equal to that of the fundamental alignment.   (In Visual C++, the fundamental alignment is the alignment that's required for a double, or 8 bytes. In code that targets 64-bit platforms, it's 16 bytes.)

[docs](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/malloc?view=msvc-170).

Are you on a 32-bit platform?

Reply:
Hi @cantonios, working on it, and will have an update soon. I am on 64-bit platform

Reply:
The get()function in the test returns a pointer to the class ManualConstructor member space_. defined in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/lib/gtl/manual_constructor.h#L238C3-L238C32 and which is forcing space_ to have the same alignment as long double. long double has 8-byte alignment on CLANG & MSVC compiler on Windows. However, long double is 16-byte aligned on Linux on Clang

The code below can reproduce the results of alignments on different compilers

#include<iostream>
 
int main( int argc, char* argv[] )

{
 
#if defined(_WIN32)

  std::cout << "Alignment of long double on Windows is " <<__alignof(long double) <<std::endl;

#else
  std::cout << "Alignment of long double on non Windows is " << __alignof__(long double) <<std::endl;

#endif
 
}  // main()

Reply:
Can we fix the tests by checking the alignment of long double to 8-byte on Windows?

Reply:
I'll need to double-check, but I believe what you want is `alignof(std::max_align_t)`

Reply:
When building TF on windows (with default MSVC) I am getting:
```
alignof(long double) == 8
alignof(std::max_align_t) == 8
alignof(void*) == 8
```
corroborating that the alignment in the manual constructor test would also be 8.

However, MSVC doesn't define the macro `__x86_64__`, so that part of the test is skipped anyway.

Is your error with using clang on windows?  I double-checked, and with `x86_64-pc-windows-gnu`, it produces 16-byte alignment, but with `x86_64-pc-windows-msvc` it produces 8-byte alignment.

Reply:
... and with `-target x86_64-pc-windows-msvc -mlong-double-80`, we have
```
alignof(long double) == 16
alignof(std::max_align_t) == 8
```
which seems wrong.